app/components/elements/forms/select_date_field_component.rb

```diff
@@ -2,10 +2,23 @@
 
 module Elements
   module Forms
-    # Component for form month fields
-    # Note that this uses select_month, which is not called on a form builder instance.
+    # Component for form date fields (e.g., day, month)
+    #
+    # Note that this uses various select_* methods from https://api.rubyonrails.org/classes/ActionView/Helpers/DateHelper.html,
+    # which are not called on a form builder instance.
+    #
     # Thus, some of the machinery of the form builder is provided below.
-    class SelectMonthFieldComponent < FieldComponent
+    class SelectDateFieldComponent < FieldComponent
+      def initialize(unit, **)
+        @unit = unit
+
+        super(**)
+      end
+
+      def select_unit(...)
+        helpers.public_send(:"select_#{@unit}", ...)
```

Comment:
I don't love this. I would never guess or get from the name that there is a single component that can render a month or day.

Would it be clearer to DRY this up with component inheritance, but leave separate components for month and day?

Reply:
@justinlittman 💬 
> I don't love this. I would never guess or get from the name that there is a single component that can render a month or day.

Yeah, I buy that. I'll ditch that refactoring.

> Would it be clearer to DRY this up with component inheritance, but leave separate components for month and day?

Maybe? I don't think it's egregiously repetitive. I'm fine with leaving it be and seeing if a better refactoring is clearer later on.